assets/pages/new-contributions-landing/contributionsLandingActions.js

```diff
@@ -116,80 +127,78 @@ const setupRegularPayment = (data: PaymentFields) =>
     }
   };
 
-const executeOneOffPayment = (data: PaymentFields) =>
-  (dispatch: Dispatch<Action>, getState: () => State): void => {
-    const state = getState();
+const executeStripeOneOffPayment = (data: StripeChargeData) =>
```

Comment:
Put all the function names next to each other and notice that vocabulary is all over the place: `execute`, `handle`, `make`, `create`, ... Perhaps some consistency would make this whole module more readable.

Reply:
yes, but some of those distinctions are principled (at least on the one-off side)

The flow is:

1 Create a payment
2. Authorise a payment
3. Execute a payment

For PayPal: we do 1 clientside, they do 2, we do 3 but serverside
For Stripe: they do 1 & 2, we do 3 clientside

So from the clientside perspective, for one-off we just see "create payment" for PayPal and "execute payment" for Stripe

Reply:
This routes file should also clarify: https://github.com/guardian/payment-api/blob/master/src/main/resources/routes

Reply:
but... "handle" vs "on" could be standardised (why `onPaymentResult` but `handleCreateOneOffPayPalPaymentResponse`?). These both are about handling a response from an AJAX call

Reply:
the "make" functions (`makeStripeOneOffPaymentData` and `makeRegularPaymentRequest`) are simply about transforming one data structure to another, so maybe they could just reflect their type signatures à la Scala convention, e.g.

`stripeChargeDataFromPaymentAuthorisation`
`regularPaymentRequestFromPaymentAuthorisation`

(setting aside that they also depend on `State`)

Reply:
I actually would have moved those last two into `readerRevenueApis` if it wasn't for the fact that they depend on `State`, which makes me think they should be in a redux-related file

Reply:
And then there's `setupRegularPayment`, where you could argue for `createRegularPayment` to be consistent with `createOneOffPayPalPayment`. But I don't want to do that because "creating" a payment has specific meaning in the one-off world, and setting up a regular payment is something quite different

Reply:
Very open to ways to rationalise all this but ultimately it boils down to the inconsistency of the various payment methods and one-off vs recurring - personally I'd prefer to expose (& explain, perhaps more comments would be helpful) this rather than trying to abstract over it, at least for now, since I've got a feeling that as soon as we add more payment methods any generalisation we come up with will break